app/frontend/src/components/RatingsSlider/RatingsSlider.tsx

```diff
@@ -0,0 +1,142 @@
+import Slider from "@material-ui/core/Slider";
+import { withStyles } from "@material-ui/core/styles";
+import {
+  AMAZING,
+  NEGATIVE,
+  NEUTRAL,
+  POSITIVE,
+  RATINGS_SLIDER,
+} from "components/RatingsSlider/constants";
+import SliderLabel from "components/RatingsSlider/SliderLabel";
+import React, { useRef } from "react";
+import { theme } from "theme";
+import makeStyles from "utils/makeStyles";
+
+const StyledSlider = withStyles({
+  root: {
+    height: 8,
+    borderRadius: 25,
+    marginTop: theme.spacing(6),
+    marginBottom: theme.spacing(1),
+  },
+  track: {
+    height: 10,
+    borderRadius: 25,
+  },
+  rail: {
+    height: 10,
+    borderRadius: 25,
+  },
+  thumb: {
+    height: 20,
+    width: 20,
+    color: theme.palette.badRating.main,
+  },
+  valueLabel: {
+    left: "calc(-50% + 4px)",
+  },
+  mark: {
+    display: "none",
+  },
+  markLabel: {
+    maxWidth: "20%",
+    whiteSpace: "pre-line",
+  },
+})(Slider);
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    '& .MuiSlider-markLabel[data-index="0"]': {
+      transform: "translateX(0%)",
+    },
+    '& .MuiSlider-markLabel[data-index="3"]': {
+      transform: "translateX(10%)",
+    },
+  },
+}));
+
+const getSliderColor = (value: number) => {
+  const goodColor = theme.palette.goodRating.main.slice(1);
+  const badColor = theme.palette.badRating.main.slice(1);
+
+  const r = Math.ceil(
+    parseInt(goodColor.substring(0, 2), 16) * value +
+      parseInt(badColor.substring(0, 2), 16) * (1 - value)
+  );
+  const g = Math.ceil(
+    parseInt(goodColor.substring(2, 4), 16) * value +
+      parseInt(badColor.substring(2, 4), 16) * (1 - value)
+  );
+  const b = Math.ceil(
+    parseInt(goodColor.substring(4, 6), 16) * value +
+      parseInt(badColor.substring(4, 6), 16) * (1 - value)
+  );
```

Comment:
This looks a lot cleaner. Would it also be good to pull out the `goodRatingRed` etc. to the constants file, or is that a bad use of the constants file?

Reply:
I don't actually have a strong opinion on that one! But personally I'd lean more towards keeping it here since it is specific to this slider and unlikely to be used elsewhere?